crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -967,29 +967,27 @@ impl<C: Comments> VisitMut for ServerActions<C> {
             self.fn_decl_ident = old_fn_decl_ident;
         }
 
-        if !self.config.is_react_server_layer {
-            return;
-        }
-
-        let mut child_names = if self.should_track_names {
-            let names = take(&mut self.names);
-            self.names = current_names;
-            self.names.extend(names.iter().cloned());
-            names
-        } else {
-            take(&mut self.names)
-        };
-
         if let Some(directive) = directive {
             if !f.is_async {
                 emit_error(ServerActionsErrorKind::InlineSyncFunction {
                     span: f.span,
-                    directive,
+                    directive: directive.clone(),
                 });
+            }
+
+            let has_errors = HANDLER.with(|handler| handler.has_errors());
```

Comment:
This may catch errors from other transforms. It would be fine if it's intentional, though.

Reply:
Yeah, I think that's fine.